Neos.Neos/Classes/Service/LinkingService.php

```diff
@@ -367,7 +370,15 @@ public function createNodeUri(
         $request = $controllerContext->getRequest()->getMainRequest();
         $uriBuilder = clone $controllerContext->getUriBuilder();
         $uriBuilder->setRequest($request);
-        $action = $workspace && $workspace->isPublicWorkspace() && !$hiddenState->isHidden ? 'show' : 'preview';
+
+        if (BackendHelper::isEditMode($request) || BackendHelper::isPreviewMode($request)) {
```

Comment:
I'd prefer to move this closer to the domain. Looks like a job for the `EditPreviewModeRepository`:

```php
final class EditPreviewModeRepository
{
    public function findOneByActionRequest(ActionRequest $request): ?EditPreviewMode;
}
```

So that this check translates to 

```suggestion
        $editPreviewMode = $this->editPreviewModeRepository->findOneByActionRequest($request);
        if ($editPreviewMode) {
```

Depending on how we move forward with the `BackendHelper` itself, it should of course use the same method to retrieve the proper `EditPreviewMode` from the request.